`webgpu` feature fails to compile

**Repro steps**
Create a new crate with wgpu 0.19 as a dependency:
```toml
[dependencies]
wgpu = "0.19"
```

Build it with `cargo build --target wasm32-unknown-unknown`

It fails to build due to missing features on web_sys:
e.g.
```
error[E0412]: cannot find type `GpuBuffer` in crate `web_sys`
    --> /home/rukai/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.19.0/src/backend/webgpu.rs:3480:29
     |
3480 |     fn new(buffer: web_sys::GpuBuffer, desc: &crate::BufferDescriptor<'_>) -> Self {
     |                             ^^^^^^^^^ not found in `web_sys`
```

I could probably enable those features in web_sys myself but wgpu should really do that for me.

If I instead do:
```toml
[dependencies]
wgpu = { version = "0.19", default-features = false }
```

It will build successfully.

**Platform**
* Linux
* rust 1.75

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -5,3 +5,6 @@ xtask = "run --manifest-path xtask/Cargo.toml"
 rustflags = [
 "--cfg=web_sys_unstable_apis"
 ]
+rustdocflags = [
+"--cfg=web_sys_unstable_apis"
+]

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -56,7 +56,7 @@ env:
   RUST_BACKTRACE: full
   PKG_CONFIG_ALLOW_CROSS: 1 # allow android to work
   RUSTFLAGS: --cfg=web_sys_unstable_apis -D warnings
-  RUSTDOCFLAGS: -Dwarnings
+  RUSTDOCFLAGS: --cfg=web_sys_unstable_apis -D warnings
   WASM_BINDGEN_TEST_TIMEOUT: 300 # 5 minutes
   CACHE_SUFFIX: c # cache busting
 

--- a/wgpu/src/backend/mod.rs
+++ b/wgpu/src/backend/mod.rs
@@ -1,9 +1,23 @@
-#[cfg(webgpu)]
+#[cfg(all(webgpu, web_sys_unstable_apis))]
 mod webgpu;
-#[cfg(webgpu)]
+#[cfg(all(webgpu, web_sys_unstable_apis))]
 pub(crate) use webgpu::{get_browser_gpu_property, ContextWebGpu};
+
+#[cfg(all(webgpu, not(web_sys_unstable_apis)))]
+compile_error!(
+    "webgpu feature used without web_sys_unstable_apis config:
+Here are some ways to resolve this:
+* If you wish to use webgpu backend, create a .cargo/config.toml in the root of the repo containing:
+    [build]
+    rustflags = [ \"--cfg=web_sys_unstable_apis\" ]
+    rustdocflags = [ \"--cfg=web_sys_unstable_apis\" ]
+* If you wish to disable webgpu backend and instead use webgl backend, change your wgpu Cargo.toml entry to:
+    wgpu = { version = \"\", default-features = false, features = [\"webgl\"] }
+"
+);
 
 #[cfg(wgpu_core)]
 mod wgpu_core;
+
 #[cfg(wgpu_core)]
 pub(crate) use wgpu_core::ContextWgpuCore;

--- a/wgpu/src/lib.rs
+++ b/wgpu/src/lib.rs
@@ -1812,7 +1812,7 @@ impl Instance {
             );
         }
 
-        #[cfg(webgpu)]
+        #[cfg(all(webgpu, web_sys_unstable_apis))]
         {
             let is_only_available_backend = !cfg!(wgpu_core);
             let requested_webgpu = _instance_desc.backends.contains(Backends::BROWSER_WEBGPU);
@@ -3086,7 +3086,7 @@ impl<'a> BufferSlice<'a> {
     /// this function directly hands you the ArrayBuffer that we mapped the data into in js.
     ///
     /// This is only available on WebGPU, on any other backends this will return `None`.
-    #[cfg(webgpu)]
+    #[cfg(all(webgpu, web_sys_unstable_apis))]
     pub fn get_mapped_range_as_array_buffer(&self) -> Option<js_sys::ArrayBuffer> {
         self.buffer
             .context

